File: 21-  Merge Two Sorted Lists/main.cs
/**
 * Definition for singly-linked list.
 * public class ListNode {
 *     public int val;
 *     public ListNode next;
 *     public ListNode(int val=0, ListNode next=null) {
 *         this.val = val;
 *         this.next = next;
 *     }
 * }
 */
public class Solution {
    public ListNode MergeTwoLists(ListNode list1, ListNode list2) {
        if (list1 is null) return list2;
        if (list2 is null) return list1;

        ListNode head;

        if (list1.val <= list2.val) {
            head = list1;
            list1 = list1.next;
        }
        else {
            head = list2;
            list2 = list2.next;
        }

        ListNode cur = head;

        while(list1 is not null && list2 is not null){
            if(list1.val <= list2.val){
                cur.next = list1;
                list1=list1.next;
            }
            else{
                cur.next = list2;
                list2=list2.next;
            }
            cur = cur.next;
        }

        if(list1 is null){
            cur.next = list2;
        }
        else{
            cur.next = list1;
        }

        return head;
    }
}

File: 21-  Merge Two Sorted Lists/main.py
# Definition for singly-linked list.
# class ListNode:
#     def __init__(self, val=0, next=None):
#         self.val = val
#         self.next = next
class Solution:
    def mergeTwoLists(self, list1: Optional[ListNode], list2: Optional[ListNode]) -> Optional[ListNode]:
        if list1 == None:
            return list2
        if list2 == None:
            return list1
        
        ordered = []
        connect = None
        temp1 = list1
        temp2 = list2
        while(temp1 != None or temp2 != None):
            if(temp1 != None):
                if(temp1.next == None):
                    connect = temp1
                ordered.append(temp1.val)
                temp1 = temp1.next
            if(temp2 != None):
                ordered.append(temp2.val)
                temp2 = temp2.next
                
        ordered.sort()
        connect.next = list2
        temp1 = list1
        count = 0
        while(temp1 != None):
            temp1.val = ordered[count]
            temp1 = temp1.next
            count += 1
            
        return list1
            
            
        